scripts/interactive_tests/interactive_test_util.js

```diff
@@ -62,15 +72,20 @@ var Client = function(port) {
   // Send a command to the server and wait for a response. Return response as a 
   // promise. 
   this.sendCommand = function(cmd) {
-    var deferred = q.defer();
+    let deferredResolve;
```

Comment:
How about wrapping everything within a `new Promise` here? It would remove the need to create our own deferred (which is a bit of an anti-pattern IHMO). 

If we must go the deferred route, we should create a helper for this pattern so we aren't repeating ourselves in multiple files.